From a48412bfa45a7996f5147d6d791f18c8b96d4103 Mon Sep 17 00:00:00 2001
From: gayan <gayan@puppet.gayan.org>
Date: Wed, 6 Aug 2014 17:20:57 +0530
Subject: [PATCH 11/30] updates

---
 .../metadataservice/listener/TopologyAgent.java    | 94 ++--------------------
 1 file changed, 5 insertions(+), 89 deletions(-)

diff --git a/components/org.apache.stratos.metadataservice/src/main/java/org/apache/stratos/metadataservice/listener/TopologyAgent.java b/components/org.apache.stratos.metadataservice/src/main/java/org/apache/stratos/metadataservice/listener/TopologyAgent.java
index 0a30bbd..b8fe918 100644
--- a/components/org.apache.stratos.metadataservice/src/main/java/org/apache/stratos/metadataservice/listener/TopologyAgent.java
+++ b/components/org.apache.stratos.metadataservice/src/main/java/org/apache/stratos/metadataservice/listener/TopologyAgent.java
@@ -23,19 +23,13 @@ package org.apache.stratos.metadataservice.listener;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.apache.stratos.messaging.event.Event;
-import org.apache.stratos.messaging.event.topology.CompleteTopologyEvent;
-import org.apache.stratos.messaging.event.topology.MemberActivatedEvent;
-import org.apache.stratos.messaging.event.topology.MemberStartedEvent;
 import org.apache.stratos.messaging.event.topology.MemberSuspendedEvent;
 import org.apache.stratos.messaging.event.topology.MemberTerminatedEvent;
-import org.apache.stratos.messaging.listener.topology.ClusterCreatedEventListener;
-import org.apache.stratos.messaging.listener.topology.CompleteTopologyEventListener;
-import org.apache.stratos.messaging.listener.topology.MemberActivatedEventListener;
-import org.apache.stratos.messaging.listener.topology.MemberStartedEventListener;
 import org.apache.stratos.messaging.listener.topology.MemberSuspendedEventListener;
 import org.apache.stratos.messaging.listener.topology.MemberTerminatedEventListener;
 import org.apache.stratos.messaging.message.receiver.topology.TopologyEventReceiver;
 import org.apache.stratos.messaging.message.receiver.topology.TopologyManager;
+import org.apache.stratos.metadataservice.services.MetaDataAdmin;
 
 /**
  * Cartridge agent runnable.
@@ -62,40 +56,6 @@ public class TopologyAgent implements Runnable {
 			log.debug("Starting topology event message receiver thread");
 		}
 		TopologyEventReceiver topologyEventReceiver = new TopologyEventReceiver();
-		topologyEventReceiver.addEventListener(new MemberActivatedEventListener() {
-			@Override
-			protected void onEvent(Event event) {
-				try {
-					log.info("Member activated event received");
-					TopologyManager.acquireReadLock();
-					if (log.isDebugEnabled()) {
-						log.debug("Member activated event received");
-					}
-					MemberActivatedEvent memberActivatedEvent = (MemberActivatedEvent) event;
-					// extensionHandler.onMemberActivatedEvent(memberActivatedEvent);
-				} catch (Exception e) {
-					if (log.isErrorEnabled()) {
-						log.error("Error processing member activated event", e);
-					}
-				} finally {
-					TopologyManager.releaseReadLock();
-				}
-			}
-		});
-		topologyEventReceiver.addEventListener(new ClusterCreatedEventListener() {
-			@Override
-			protected void onEvent(Event event) {
-				try {
-					log.info("Event received: " + event);
-
-				} catch (Exception e) {
-					log.error("Error processing event", e);
-				} finally {
-					TopologyManager.releaseReadLock();
-				}
-			}
-
-		});
 
 		topologyEventReceiver.addEventListener(new MemberTerminatedEventListener() {
 			@Override
@@ -107,7 +67,10 @@ public class TopologyAgent implements Runnable {
 						log.debug("Member terminated event received");
 					}
 					MemberTerminatedEvent memberTerminatedEvent = (MemberTerminatedEvent) event;
-					// extensionHandler.onMemberTerminatedEvent(memberTerminatedEvent);
+					System.out.println("Terminated event :::::::::::::::::::: " +
+					                   memberTerminatedEvent.getServiceName());
+					new MetaDataAdmin().removeCartridgeMetaDataDetails("appA", "php");
+
 				} catch (Exception e) {
 					if (log.isErrorEnabled()) {
 						log.error("Error processing member terminated event", e);
@@ -139,53 +102,6 @@ public class TopologyAgent implements Runnable {
 			}
 		});
 
-		topologyEventReceiver.addEventListener(new CompleteTopologyEventListener() {
-			private boolean initialized;
-
-			@Override
-			protected void onEvent(Event event) {
-				if (!initialized) {
-					try {
-						log.info("Complete topology event received");
-						TopologyManager.acquireReadLock();
-						if (log.isDebugEnabled()) {
-							log.debug("Complete topology event received");
-						}
-						CompleteTopologyEvent completeTopologyEvent = (CompleteTopologyEvent) event;
-						// extensionHandler.onCompleteTopologyEvent(completeTopologyEvent);
-						initialized = true;
-					} catch (Exception e) {
-						if (log.isErrorEnabled()) {
-							log.error("Error processing complete topology event", e);
-						}
-					} finally {
-						TopologyManager.releaseReadLock();
-					}
-				}
-			}
-		});
-
-		topologyEventReceiver.addEventListener(new MemberStartedEventListener() {
-			@Override
-			protected void onEvent(Event event) {
-				try {
-					log.info("Member started event received");
-					TopologyManager.acquireReadLock();
-					if (log.isDebugEnabled()) {
-						log.debug("Member started event received");
-					}
-					MemberStartedEvent memberStartedEvent = (MemberStartedEvent) event;
-					// extensionHandler.onMemberStartedEvent(memberStartedEvent);
-				} catch (Exception e) {
-					if (log.isErrorEnabled()) {
-						log.error("Error processing member started event", e);
-					}
-				} finally {
-					TopologyManager.releaseReadLock();
-				}
-			}
-		});
-
 		Thread thread = new Thread(topologyEventReceiver);
 		thread.start();
 		if (log.isDebugEnabled()) {
-- 
1.9.1

